src/components/NewCommentForm.tsx

```diff
@@ -47,25 +46,31 @@ export const NewCommentForm: React.FC<Props> = ({ onSubmit }) => {
   const handleSubmit = async (event: React.FormEvent) => {
     event.preventDefault();
 
-    setErrors({
-      name: !name,
-      email: !email,
-      body: !body,
-    });
+    if (selectedPost) {
+      const newComment = {
+        name: name.trim(),
+        email: email.trim(),
+        body: body.trim(),
+        postId: selectedPost?.id,
+      };
 
-    if (!name || !email || !body) {
-      return;
-    }
+      setErrors({
+        name: !newComment.name,
+        email: !newComment.email,
+        body: !newComment.body,
+      });
+
+      if (!newComment.name || !newComment.email || !newComment.body) {
+        return;
+      }
 
-    setSubmitting(true);
+      setSubmitting(true);
 
-    // it is very easy to forget about `await` keyword
-    await onSubmit({ name, email, body });
+      await dispatch(addNewComment(newComment));
```

Comment:
Suggestion: Consider handling potential errors from the `dispatch` call. If `addNewComment` fails, it might be useful to inform the user or log the error.